Assets/MRTK/Core/Providers/BaseGlobalInputSource.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Linq;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Base class for input sources whose pointers are all active pointers in the scene.
+    /// </summary>
+    /// <remarks>
+    /// <para>This base class is intended to represent input sources which raise event to all active pointers found by the FocusProvider in a scene.</para>
+    /// </remarks>
+    public class BaseGlobalInputSource : IMixedRealityInputSource, IDisposable
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public BaseGlobalInputSource(string name, IMixedRealityFocusProvider focusProvider, InputSourceType sourceType = InputSourceType.Other)
+        {
+            SourceId = (CoreServices.InputSystem != null) ? CoreServices.InputSystem.GenerateNewSourceId() : 0;
+            SourceName = name;
+            FocusProvider = focusProvider;
+            SourceType = sourceType;
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; }
+
+        /// <inheritdoc />
+        public string SourceName { get; }
+
+        /// <inheritdoc />
+        public virtual IMixedRealityPointer[] Pointers { get; set; }
+
+        /// <inheritdoc />
+        public InputSourceType SourceType { get; set; }
+
+        private IMixedRealityFocusProvider FocusProvider;
+
+        public void UpdateActivePointers()
+        {
+            Pointers = FocusProvider.GetPointers<IMixedRealityPointer>().Where(x => x.IsActive).ToArray();
```

Comment:
yeah I was considering refreshing everytime it's called, but I think that would cause a noticeable perf hit given we're resorting to Linq methods to filter things out the inactive pointers. Given that the pointers can change their state, I don't think there's currently a better way to find all of the active pointers without having to rely on specific aspects of the FocusProvider implementation